Jenkinsfile

```diff
@@ -214,6 +226,100 @@ stage('Sanity Check') {
   }
 }
 
+def build_image(image_name) {
+  hash = sh(
+    returnStdout: true,
+    script: 'git log -1 --format=\'%h\''
+  ).trim()
+  hash = "${env.BRANCH_NAME}-${hash}"
+  sh(
+    script: "${docker_build} ${image_name} --spec ${image_name}:${hash}",
+    label: 'Building docker image'
+  )
+  sh "echo NYI: Uploading docker image to registry..."
+}
+
+if (rebuild_docker_images) {
+  stage('Docker') {
+    // TODO in a follow up PR: Upload to ECR, find tag and use in
+    // subsequent builds
+    parallel 'ci-lint': {
```

Comment:
do we have to do all of them?

Reply:
yes but since these are pretty infrequent I'll leave that for a follow up, I added an action item to #10646